eventhandlers/action_runners/actions/VertexTrainDatasetAction.py

```diff
@@ -0,0 +1,291 @@
+                outfile.write('\n')
+
+        gcp_data_tools.upload_to_cloud_storage(f"{temp_folder_name}/{filename}", f"{self.action.config_data.get('model_name')}/{filename}")
+        self.clean_up_temp_file(f"{temp_folder_name}/{filename}")
+
+        return f"gs://{self.action.config_data.get('staging_bucket_name_without_gs_prefix')}/{self.action.config_data.get('model_name')}/{filename}"
+
+    def init_ai_platform(self, google_vertex_connector):
+        aiplatform.init(
+            credentials = google_vertex_connector.get_credentials(),
+            project = google_vertex_connector.auth_data['project_id'],
+            location = "us-central1",
+            staging_bucket = 'gs://' + self.action.config_data.get('staging_bucket_name_without_gs_prefix'),
+            experiment = self.action.config_data.get('experiment'),
+            experiment_description = self.action.config_data.get('experiment_description')
+        )
+
+    def init_gcp_data_tools(self, google_vertex_connector):
+        bucket_config = {
+            "GOOGLE_PROJECT_NAME": self.action.config_data.get('gcp_project_id'),
+            "CLOUD_STORAGE_BUCKET": self.action.config_data.get('staging_bucket_name_without_gs_prefix'),
+            "ML__CLOUD_STORAGE_BUCKET": self.action.config_data.get('staging_bucket_name_without_gs_prefix'),
+            "credentials": google_vertex_connector.get_credentials()
+        }
+
+        gcp_data_tools = DataToolsGCP(bucket_config)
+
+        return gcp_data_tools
+
+    def write_diffgram_blob_to_gcp(self, gcp_data_tools, temp_folder_name, file):
+        blob_bytes = data_tools.download_bytes(file.image.url_signed_blob_path)
+        filename = file.image.original_filename
+
+        try:
+            os.makedirs(temp_folder_name)
+        except OSError:
+            pass
+
+        with open(f"{temp_folder_name}/{filename}", "wb") as binary_file:
+            binary_file.write(blob_bytes)
+
+        gcp_data_tools.upload_to_cloud_storage(f"{temp_folder_name}/{filename}", f"{self.action.config_data.get('model_name')}/{filename}", "image")
+        self.clean_up_temp_file(f"{temp_folder_name}/{filename}")
+
+        return f"{self.action.config_data.get('model_name')}/{filename}"
+
+    def count_dataset_labels(self, session, file_list):
+        label_count = {}
+
+        for file in file_list:
+            instance_list = Instance.list(session=session, file_id=file['diffgram_file'].id)
+            for instance in instance_list:
+                label = File.get_by_id(session=session, file_id=instance.label_file_id)
+                if label is not None:
+                    if label_count.get(label.label.name) == None:
+                        label_count[label.label.name] = 1
+                    else:
+                        label_count[label.label.name] += 1
+        
+        self.label_count = label_count
+        return label_count
+
+    def train_automl_model(self, credentials, dataset):
```

Comment:
Some of these function can be moved to the VertexAI connector.

For example: 
* `train_automl_model`
* `write_diffgram_blob_to_gcp`
* `init_ai_platform`

Reply:
I'm not sure if we actually need a Vertex AI connector. As I understand, connector is more like reusable thing that we may use in different parts on Diffgram, but Vertex AI - is more specific to the action, no?